e2e/specs/results-viewer.spec.ts

```diff
@@ -0,0 +1,189 @@
+    await page.locator('#manualEntryAlkalinePhosphastaseid').fill('10');
+    await page.locator('#manualEntrySerumUricAcidid').fill('10');
+    await page.locator('#manualEntryTotalCholesterolid').fill('10');
+    await page.locator('#manualEntryTriglyceridesid').fill('10');
+    await page.locator('#manualEntryAmylaseid').fill('10');
+    await page.locator('#manualEntrySerumCarbonDioxideid').fill('10');
+  });
+
+  await test.step('When I click the `Other` tab', async () => {
+    await page.getByRole('button', { name: 'Other', exact: true }).click();
+  });
+
+  await test.step('And I fill the other test results', async () => {
+    await page.locator('#manualEntryCultureandSensitivityUrineid').fill('83%');
+    await page.locator('#manualEntryUrinePregnancyTestid_1').check();
+    await page.locator('#manualEntryUrineProteinDipStickid_0').check();
+    await page.locator('#manualEntryUrineBacteriuriaTestid_1').check();
+    await page.locator('#manualEntryErythrocytesPresenceInUrineSedimentByLightMicroscopyTestid_1').check();
+    await page.locator('#manualEntryLeukocytesPresenceInUrineSedimentByLightMicroscopyid_2').check();
+    await page.locator('#manualEntryEpithelialCastsPresenceInUrineSedimentByLightMicroscopyTestid_2').check();
+    await page.locator('#manualEntryYeastPresenceInUrineSedimentByLightMicroscopyid_1').check();
+    await page.locator('#manualEntrySporePresenceInUrineTestid_3').check();
+    await page.locator('#manualEntryTrichomonasVaginalisPresenceInUrineSedimentByLightMicroscopyid_0').check();
+    await page.locator('#manualEntryCrystalsTypeInUrineSedimentByLightMicroscopyTestid_3').check();
+    await page.locator('#manualEntryStoolCultureBacterialid').fill('no signs');
+    await page.locator('#manualEntryStoolExamid').selectOption('Cestode Infection');
+    await page.locator('#manualEntryStoolMicroscopyWithConcentrationid').fill('no signs');
+    await page.locator('#manualEntryKinyounsStainForCoccidiansid').fill('no signs');
+    await page.locator('#manualEntryStoolFatTestSemiQuantitativeid_3').check();
+    await page.locator('#manualEntryStoolTestforReducingSubstanceid_0').check();
+    await page.locator('#manualEntryFecalOccultBloodTestid_2').check();
+  });
+
+  await test.step('And I save the form', async () => {
+    await page.getByRole('button', { name: /save and close/i }).click();
+  });
+
+  await test.step('Then I should see a success toast notification', async () => {
+    await expect(page.getByText('The form has been submitted successfully.')).toBeVisible();
+  });
+
+  await test.step('When I reload the viewer page', async () => {
+    await resultsViewerPage.goTo(patient.uuid);
+  });
+
+  await test.step('Then I should see 12 and 15 as white and red blood cells test results displayed respectively', async () => {
+    await expect(page.getByRole('link', { name: /white blood cells/i })).toBeVisible();
+    await expect(page.getByText('12', { exact: true })).toBeVisible();
```

Comment:
How does the test know that `12` is a value for white blood cell?

Reply:
@kdaud
AFAIK, given the HTML code, there is no perfect way to tie the `12` to white blood cells.
I just made sure only white blood cell was assigned `12` and I'm using `exact` so we don't find the whole word `12` elsewhere on the page.

Why did I approach it that way?
See the grid code below, I am not sure how we can perfectly make the connection between the test and it's result during the test. I will appreciate any guide on this. 
![image](https://github.com/openmrs/openmrs-esm-patient-chart/assets/19818300/1c41606d-870e-46fb-b63a-203f2c649abe)

What I can do further is to ensure the filter checkbox is used to view only the white blood cells. Though it doesn't change the current method am using, it just helps to streamline what's displayed and also tests the filter.

![image](https://github.com/openmrs/openmrs-esm-patient-chart/assets/19818300/b9560fa2-1767-47d5-9501-3d5e24479f88)

Your thoughts. please.

Reply:
We should probably modify the JSX to use more semantic HTML and accessibility roles, but that can come as an enhancement on top of this work.